src/shopify/theme_client.go

```diff
@@ -314,6 +314,41 @@ func (c Client) UpdateAsset(asset Asset) error {
 	return nil
 }
 
+// UpdateWatchedAsset will take an asset and will return when the asset has been updated.
+// If there was an error, in the request then error will be defined otherwise the
+//response will have the appropropriate data for usage.
+func (c Client) UpdateWatchedAsset(asset Asset, oldChecksum string) error {
```

Comment:
Currently this heavily duplicates the existing `UpdateAsset` but I should be able to extract out the commonality.

Reply:
Why not just update the `UpdateAsset` method call and only add the header if it is non-empty.